minari/dataset/minari_dataset.py

```diff
@@ -162,24 +162,18 @@ def __init__(
             action_space=self._data.action_space,
             data_path=str(self._data.data_path),
         )
-        self._total_steps = None
+        self._total_steps = total_steps
         self._generator = np.random.default_rng()
 
     @property
     def total_episodes(self):
         """Total episodes recorded in the Minari dataset."""
         assert self._episode_indices is not None
-        return len(self._episode_indices)
+        return self._episode_indices.size
 
     @property
     def total_steps(self):
         """Total episodes steps in the Minari dataset."""
-        if self._total_steps is None:
-            t_steps = self._data.apply(
-                lambda episode: episode.total_steps,
-                episode_indices=self._episode_indices,
-            )
-            self._total_steps = sum(t_steps)
         return self._total_steps
```

Comment:
I would totally remove the function and change the name of the attribute `_total_steps` to `total_steps` 
[It is not pythonic to have private attribute with getter without a good reason]